### PR TITLE
fix: firstKeyPos cmdSlot

### PR DESCRIPTION
**Pull request by jamsonzan**
#1501 

only **xread xreadgroup zunionstore zinterstore** need to fix

### CODE REVIEW

File: command.go [hunk 1]
```diff
@@ -1976,14 +1987,14 @@ func (cmd *GeoPosCmd) readReply(rd *proto.Reader) error {
...
-	StepCount   int8
-	ReadOnly    bool
+	Name         string
+	Arity        int8
+	Flags        []string
+	ACLFlags     []string
+	FirstKeyBase int8
+	LastKeyBase  int8
+	StepCount    int8
+	ReadOnly     bool
```
Comment (vmihailenco) [inline command.go:1997]: Please revert - there is no need to change this to fix the issue.

File: command.go [hunk 2]
```diff
@@ -106,6 +108,7 @@ type baseCmd struct {
 	ctx  context.Context
 	args []interface{}
 	err  error
+	keyOffset int
```
Comment (vmihailenco) [inline command.go:111]: Using `keyBase+keyOffset` looks a bit complicated. Can't we rename this to `keyPos` and use it when `cmd.keyPos != 0` and fallback to `info.FirstKeyPos`?

File: command.go [hunk 3]
```diff
@@ -76,7 +78,7 @@ func cmdFirstKeyPos(cmd Cmder, info *CommandInfo) int {
 	if info == nil {
 		return 0
 	}
-	return int(info.FirstKeyPos)
+	return int(info.FirstKeyBase) + cmd.firstKeyOffset()
```
Comment (vmihailenco) [inline command.go:81]: E.g.

```
if pos := cmd.firstKeyPos(); pos != 0 {
    return pos
}
return info.FirstKeyPos
```

Comment (jamsonzan) [reply command.go]: good!

**Comment by vmihailenco** [comment]
Thanks!

**Pull request merged**